### ISSUE TITLE
Unable to create OpenGL context

### ISSUE THREAD

**Issue by unegare** [PacktPublishing/OpenGL-4-Shading-Language-Cookbook-Third-Edition#3]
```
$ ./chapter01 basic
Unable to create OpenGL context.
$./chapter02 basic-attrib
Unable to create OpenGL context.
$ ./chapter03 diffuse
Unable to create OpenGL context.
```

In spite of success executions (but with a segfault sometimes) of examples from [the 2nd edition](https://github.com/daw42/glslcookbook) of this book.

**Comment by daw42** [comment]
Are you on Windows?  If so, does your graphics driver support OpenGL 4.6?  The 2nd edition only required OpenGL 4.3.

**Comment by unegare** [comment]
No, I'm on Ubuntu 18.04. gtx 1060.
OpenGL Information:
Version: 4.6.0 NVIDIA 390.116

**Comment by daw42** [comment]
Ok, a couple things to try:  if you're using an older version of GLFW, you might try updating it.  If that doesn't work, look at `ingredients/scenerunner.h`.  It has all of the context initialization stuff.  You can try changing some of the parameters there to see what might be causing it to be unable to create a context.

**Comment by daw42** [comment]
Closing due to lack of response.

**Comment by thatrobotguy** [comment]
I am having this same issue on Windows 10. I know I have OpenGL 4.6 with NVidia driver 44X.XX installed, just wondering what other people have done in these scenarios.

**Comment by daw42** [comment]
You can try changing the values here: https://github.com/PacktPublishing/OpenGL-4-Shading-Language-Cookbook-Third-Edition/blob/a37a89b452e6078e1572eeaf97d277285da824ad/ingredients/scenerunner.h#L33

```c++
glfwWindowHint( GLFW_CONTEXT_VERSION_MAJOR, 4 );
glfwWindowHint( GLFW_CONTEXT_VERSION_MINOR, 6 );
```

Trying lower minor versions may help.  If that doesn't, you can try commenting out this line:

https://github.com/PacktPublishing/OpenGL-4-Shading-Language-Cookbook-Third-Edition/blob/a37a89b452e6078e1572eeaf97d277285da824ad/ingredients/scenerunner.h#L35

**Comment by thatrobotguy** [comment]
Technically, that code fix was exactly what was needed.
It turns out that the issue was that I requested a version of `OpenGL` that I did not have in my C++ files. It turns out that the `vcpkg` package manager defaults `OpenGL` to version `3.2`, not `4.X`. I will have to look for other options in `vcpkg` or import it in a different way entirely. I know my machine supports `4.6`, but I am not requesting the proper version. I am considering this issue closed for me.
Thanks for the help @daw42.
